### PR TITLE
Bug 1906304: Fix NaN in Memory/CPU pod charts

### PR DESCRIPTION
**Pull request by rawagner**
@kyoto @rebeccaalpert can you try if it fixes the issue for you ? Seems to work fine for me.

### CODE REVIEW

**Comment by rebeccaalpert** [comment]
@rawagner - I'm still seeing it. I had tried this earlier and it didn't seem to fix it. I pulled down your PR and the issue is still showing up intermittently for me. If you're not seeing it, just keep refreshing and trying to use the tooltips; it doesn't happen every time.

**Comment by rawagner** [comment]
@rebeccaalpert I've updated the fix. Can you try again ?

**Comment by spadgett** [comment]
/override ci/prow/ceph-storage-plugin

The ceph-storage-plugin job should only have run on PRs with changes to `frontend/packages/ceph-storage-plugin`. It failed due to an unrelated CI config change. Overriding status on this PR.

**Review by rebeccaalpert — APPROVED** [review]
I haven't seen the issue since you made these updates! Thanks.

**Comment by spadgett** [comment]
@rawagner FYI, needs rebase

**Comment by rawagner** [comment]
rebased

**Review by kyoto — COMMENTED** [review]
After looking at this for a while, I'm not sure which code change(s) are actually addressing the bug(?), but I did some testing and it does seem to be fixed.

I recently reworked the `QueryBrowser` tooltips which means they now behave more like these tooltips. I think the approach I ended up with there (basically, using custom SVG) is simpler than the approach here. In the future, it's probably worth having a go at generalizing the `QueryBrowser` tooltips so they can be used for this component too.

File: frontend/public/components/graphs/area.tsx [hunk 1]
```diff
@@ -171,7 +178,7 @@ export const AreaChart: React.FC<AreaChartProps> = ({
 };
 
 export const Area: React.FC<AreaProps> = ({
-  endTime,
+  endTime = Date.now(),
```
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/public/components/graphs/area.tsx:181]: this fixes an issue where multiple queries (utilization, limits, requests) are not synced with their end/start time.

File: frontend/public/components/graphs/area.tsx [hunk 2]
```diff
@@ -120,11 +120,18 @@ export const AreaChart: React.FC<AreaChartProps> = ({
           activateData={false}
           cursorDimension="x"
           labels={(props) => getLabel(props, false)}
+          mouseFollowTooltips
```
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/public/components/graphs/area.tsx:123]: interesting, thanks for the insights :)

File: frontend/public/components/graphs/area.tsx [hunk 3]
```diff
@@ -120,11 +120,18 @@ export const AreaChart: React.FC<AreaChartProps> = ({
...
           labelComponent={
             <ChartLegendTooltip
               stack={showAllTooltip}
               legendData={legendData}
-              title={(d) => (showAllTooltip ? formatDate(d[0].x) : getLabel({ datum: d[0] }))}
+              title={(d) => {
+                if (showAllTooltip) {
+                  return formatDate(d[0].x);
+                }
+                const usageDatum = d.find((datum) => datum.childName === 'usage');
```
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/public/components/graphs/area.tsx]: removed the knowledge about `usage`. The childName can be optionally passed from parent component.

File: frontend/public/components/graphs/tooltip.tsx [hunk 4]
```diff
@@ -232,18 +234,29 @@ export const ChartLegendTooltip: React.FunctionComponent<ChartLegendTooltipProps
...
+      const usageIndex = legendTooltipProps.legendData.findIndex((d) => d.name === 'usage');
+      if (usageIndex !== -1) {
+        const tooltipText = [...legendTooltipProps.text];
+        tooltipText.splice(usageIndex, 1),
+          (sizeProps = {
+            ...legendTooltipProps,
+            text: tooltipText,
+          });
+      }
+    }
```
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/public/components/graphs/tooltip.tsx]: simpler is always better! Updated.

Im all in for using the `QueryBrowser` tooltips in the future

**Comment by rawagner** [comment]
@kyoto implemented all of your feedback

**Review by spadgett — APPROVED** [review]
/lgtm

File: frontend/public/components/graphs/area.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kyoto) [reply frontend/public/components/graphs/area.tsx:178]: Ah OK. Thanks for the explanation!

File: frontend/public/components/graphs/area.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (kyoto) [reply frontend/public/components/graphs/area.tsx]: 👍 Thanks

**Pull request merged**